explain-subqueries.md

```diff
@@ -151,3 +151,126 @@ EXPLAIN SELECT * FROM t3 WHERE t1_id NOT IN (SELECT id FROM t1 WHERE int_col < 1
++------------------------------+---------+-----------+---------------+-----------------------------------------------------------------------------------------------------+
+8 rows in set (0.01 sec)
+```
+
+In the first query statement `EXPLAIN SELECT (a,b) IN (SELECT * FROM s) FROM t;`, since the `a` and `b` columns of table `t` and `s` are NULLABLE, the left outer semi join converted by `IN` subquery is null-aware. The implementation detail is: calculate the Cartesian product first, then put the column connected by `IN` or `= ANY` into other conditions as normal equality query for filtering.
+
+In the second query statement `EXPLAIN SELECT * FROM t WHERE (a,b) IN (SELECT * FROM s);`, since the `a` and `b` columns of table `t` and `s` are NULLABLE, the `IN` subquery should be converted to a null-aware semi join. But TiDB optimizes it to convert semi join to inner join and aggregate directly. This is because `NULL` and `false` are equivalent in `IN` subqueries for non-scalar output. The `NULL` rows in the push-down filter results in the negative semantics of the `WHERE` clause. Therefore, these rows can be ignored beforehand.
+
+> **Note:**
+>
+> The `Exists` operator is also converted to semi join, but the `Exists` operator itself is not null-aware.
+
+## Null-aware anti semi join (`NOT IN` and `!= ALL` subquery)
+
+The value of `NOT IN` and `!= ALL` set operations is three-valued (`true`, `false`, and `NULL`). For the join type converted by these subqueries, TiDB needs to be aware of the `NULL` on both sides of the Join key and process it in a special way.
+
+For `NOT IN` and `! = ALL` operators, subqueries are converted to anti semi join and anti left outer semi join separately. In the preceding example of [Anti semi join](#anti-semi-join-not-in-subquery), since the columns `test.t3.t1_id` and `test.t1.id` on both sides of the join key are `not NULL`, anti semi join does not need to be considered as null-aware (`NULL` is not processed specially).
+
+TiDB v6.3.0 optimizes for null-aware anti join (NAAJ):
+
+- Build hash join using null-aware equality condition (NA-EQ)
+
+    Set operators introduce the equality condition, which requires a special process for the `NULL` of operators on both sides of the condition. The equality condition that requires null-aware is called NA-EQ. Different from versions before TiDB v6.3.0, TiDB v6.3.0 no longer processes NA-EQ as normal EQ, but specially places it in other conditions after join, and then determines the legitimacy of the result set after matching the Cartesian product.
+
+    Since TiDB v6.3.0, NA-EQ, a weakened equality condition, is still used to build hash join. This reduces the matching amount of data that needs to be traversed and speeds up the matching process. The acceleration effect is more significant when the `DISTINCT()` value of the build table is large.
```

Comment:
```suggestion
    Since TiDB v6.3.0, NA-EQ, a weakened equality condition, is still used to build hash join. This reduces the matching amount of data that needs to be traversed and speeds up the matching process. The acceleration is more significant when the `DISTINCT()` value of the build table is large.
```